client/src/components/Search/Filters/TagsFilter/TagsFilter.js

```diff
@@ -0,0 +1,140 @@
+            return {
+                ...state.materialsFilterSearch,
+                ...state.materialsFilterLoad
+            }
+        } else {
+            return {
+                ...state.tagsFilterSearch,
+                ...state.tagsFilterLoad
+            };
+        }
+    }
+
+    componentDidMount() {
+        if (this.state.searchQuery) this.searchTags();
+        if (this.props.tags && this.state.selectedTags.length === 0) {
+            this.loadTags();
+        }
+    }
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        if ((!prevProps.selectedTags || !prevProps.selectedTags.length) &&
+            this.props.selectedTags && this.props.selectedTags.length) {
+                this.setState({
+                    selectedTags: (this.state.selectedTags || []).concat(this.props.selectedTags)
+                        .map(tag => {
+                            tag.selected = true;
+                            return tag;
+                        })
+                })
+        }
+    }
+
+    render() {
+        const { searchQuery, selectedTags } = this.state;
+        const { searchResults, variant } = this.props;
+        const visibleSearchResults = searchResults.filter(result => {
+            return !selectedTags.find(tag => tag.id === result.id);
+        });
+
+        return (
+            <div className="tags-filter">
+                <div className="selected-tags">
+                    <Tags selectable onSelect={this.handleSelectTag.bind(this)} tags={selectedTags} selectedIcon="clear"/>
+                </div>
+                <div className="search">
+                    <input type="text"
+                           value={searchQuery}
+                           onChange={(event) => this.handleSearchChange(event.target.value)}
+                           placeholder={variant.charAt(0).toUpperCase() + variant.slice(1) + '...'}
+                           className="form-control"
+                           onKeyDown={(event) => this.handleKeyDown(event)}/>
+                    {searchQuery && <i className="material-icons search-clear" onClick={() => this.handleClear()}>clear</i>}
+                </div>
+                <div className="search-results">
+                    <Tags selectable onSelect={this.handleSelectTag.bind(this)} tags={visibleSearchResults}/>
+                </div>
+            </div>
+        );
+    }
+
+    async handleSelectTag(value, tag) {
+        let { selectedTags } = this.state;
+        // If the tag is being selected, add it to the selected tags and sort them alphabetically
+        if (value) {
+            tag.selected = true;
+            selectedTags.push(tag);
+            selectedTags = selectedTags.sort();
+        // If the tag is being deselected, remove it from the selected tags
+        } else {
+            const index = selectedTags.findIndex(t => t.name === tag.name);
+            if (index >= 0) {
+                [ tag ] = selectedTags.splice(index, 1);
```

Comment:
What does the `[ ]` around `tag` do here?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Array_destructuring